internal/services/containers/kubernetes_cluster_node_pool_resource_test.go

```diff
@@ -3506,3 +3530,121 @@ resource "azurerm_kubernetes_cluster_node_pool" "pool2" {
 		data.RandomInteger,     // kubernetes_cluster dns_prefix
 	)
 }
+
+func (KubernetesClusterNodePoolResource) podIPAllocationModeConfig(data acceptance.TestData) string {
```

Comment:
Please combine the two configuration function and pass the allocation mode as an argument.